automated install/basic-install.sh

```diff
@@ -1754,8 +1754,25 @@ FTLinstall() {
   # Move into the temp ftl directory
   pushd "$(mktemp -d)" > /dev/null || { echo "Unable to make temporary directory for FTL binary download"; return 1; }
 
-  # Always replace pihole-FTL.service
-  install -T -m 0755 "${PI_HOLE_LOCAL_REPO}/advanced/pihole-FTL.service" "/etc/init.d/pihole-FTL"
+  # Determine if systemd is used on this system
+  if file "$(which init)" | grep "systemd" &> /dev/null; then
```

Comment:
The canonical check whether systemd is the active init system is
``` test -e /run/systemd/system```
See https://www.freedesktop.org/software/systemd/man/sd_booted.html

The only case I can imagine where this would fail is, if you install pihole in a chroot.

Reply:
I have actually tested running pihole on a synology NAS, I ended up just buying another pi and dedicating it to the purpose of being my DNS device.

If one was to assume that supporting pihole on synology NAS's (or really any proprietary platform) should be a desirable outcome, then there are only really two options:

1. Put a bunch of effort into make pihole run in a container (most things nowadays run docker in some form or another so that gets it into a known sandbox where it can do its thing)

2. Actually publish native packages for those proprietary environments, ie. synology NAS's basically use weirdly formed init services that are written as json things (I haven't looked particularly closes so could very well be very very wrong on this) and they are all handled by synoservicectl just to make things more difficult. So actually publishing a synology package, signing it and putting it into a synology package repo would be the most native way to do this.

Reply:
Also, probably worth noting, my synology ds716+II doesn't have `file` installed on it by default, so you can't just use file to check what the init system is on a synology nas because it just won't execute it.

You could something like:

```
if stat $(which init) | grep -q systemd; then
...
```
with the if file thing I get
```
hvindin@diskstation:~$ if file $(which init) | grep -q systemd; then echo "systemd"; else echo "not systemd"; fi
bash: file: command not found
not systemd
hvindin@diskstation:~$ if stat $(which init) | grep -q systemd; then echo "systemd"; else echo "not systemd"; fi
not systemd
```

But then I'm not sure if stat is more or less common than file... It's on all by debian, centos, rhel, synology etc. boxes but there are a shit load of distros and file might just be the more common thing to have installed by default.

But in particular on purpose built hardware being repurposed I feel like it's less likely that common nixy utilities are going to be present.